gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/task/KubernetesBuildTask.java

```diff
@@ -57,14 +57,12 @@ public void run() {
       kitLogger.info("Building container image in [[B]]Kubernetes[[B]] mode");
     }
     try {
-      for (ImageConfiguration imageConfig : resolvedImages) {
-        storeTimestamp(
-            getBuildTimestampFile(kubernetesExtension.javaProject.getBuildDirectory().getAbsolutePath(),
-                DOCKER_BUILD_TIMESTAMP),
-            getBuildTimestamp(null, null, kubernetesExtension.javaProject.getBuildDirectory().getAbsolutePath(),
-                DOCKER_BUILD_TIMESTAMP));
-        jKubeServiceHub.getBuildService().build(imageConfig);
-      }
+      storeTimestamp(
+          getBuildTimestampFile(kubernetesExtension.javaProject.getBuildDirectory().getAbsolutePath(),
+              DOCKER_BUILD_TIMESTAMP),
+          getBuildTimestamp(null, null, kubernetesExtension.javaProject.getBuildDirectory().getAbsolutePath(),
+              DOCKER_BUILD_TIMESTAMP));
+      jKubeServiceHub.getBuildService().build(resolvedImages.toArray(new ImageConfiguration[0]));
```

Comment:
what is the reason of getting only the first image configuration?

Reply:
I feel like this is fragile, could `resolvedImages` be empty?

Reply:
This is just a file with the timestamp which is reused in all images.
After many refactors this has been reduced to what you see now. It belongs somewhere else, but that should be handled in an additional refactor.

Reply:
> I feel like this is fragile, could resolvedImages be empty?

They can be empty, but can't be null